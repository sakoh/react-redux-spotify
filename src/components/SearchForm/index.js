import React, { Component } from 'react'
import Button from '@material-ui/core/Button'
import TextField from '@material-ui/core/TextField'

class SearchForm extends Component {
  state = { value: '' }

  onInput = value => this.setState({ value })

  render () {
    return (
      <form>
        <TextField label='Search Artist' onInput={this.onInput} />
        <Button>Search</Button>
      </form>
    )
  }
}

export { SearchForm }
