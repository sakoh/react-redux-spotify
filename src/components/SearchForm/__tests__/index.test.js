import React from 'react'
import { configure, shallow } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import PropTypes from 'prop-types'
import SearchForm from '..'

configure({ adapter: new Adapter() })

describe('SearchForm', () => {
  it('should render correctly', () => {
    const wrapper = shallow(<SearchForm />)

    expect(wrapper).toMatchSnapshot()
  })
})
