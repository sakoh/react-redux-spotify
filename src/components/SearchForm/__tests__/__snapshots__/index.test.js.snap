// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchForm should render correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SearchForm />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <TextField
          label="Search Artist"
          onInput={[Function]}
          required={false}
          select={false}
/>,
        <WithStyles(Button)>
          Search
</WithStyles(Button)>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "label": "Search Artist",
          "onInput": [Function],
          "required": false,
          "select": false,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": "Search",
        },
        "ref": null,
        "rendered": "Search",
        "type": [Function],
      },
    ],
    "type": "form",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <TextField
            label="Search Artist"
            onInput={[Function]}
            required={false}
            select={false}
/>,
          <WithStyles(Button)>
            Search
</WithStyles(Button)>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "label": "Search Artist",
            "onInput": [Function],
            "required": false,
            "select": false,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "Search",
          },
          "ref": null,
          "rendered": "Search",
          "type": [Function],
        },
      ],
      "type": "form",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`
